lib/schema_validations/active_record/validations.rb

```diff
@@ -133,15 +134,19 @@ def load_column_validations #:nodoc:
         end
 
         def load_integer_column_validations(name, column) # :nodoc:
-          integer_range = ::ActiveRecord::Type::Integer.new.range
-          # The Ruby Range object does not support excluding the beginning of a Range,
-          # so we always include :greater_than_or_equal_to
-          options = { :allow_nil => true, :only_integer => true, greater_than_or_equal_to: integer_range.begin }
-
-          if integer_range.exclude_end?
-            options[:less_than] = integer_range.end
+          if column.limit && column.limit > INTEGER_LIMIT
```

Comment:
Instead of using a hard-coded integer size limit (that just happens for both MySQL and Postgres), we can just calculate the range for a signed integer with the number of bytes specified by the limit.
You can easily get this range with `::ActiveRecord::Type::Integer.new(limit: column.limit).range` and ActiveRecord will take care of the rest (including using the database-default limit if `column.limit` is nil).